guppylang/prelude/_internal/compiler.py

```diff
@@ -249,6 +251,51 @@ def compile(self, args: list[Wire]) -> list[Wire]:
         return [q]
 
 
+class AngleOpCompiler(CustomCallCompiler):
+    """Compiler for tket2 angle ops.
+
+    Automatically translated between the Hugr usize type used in the angle extension
+    and Guppy's `nat` type.
+    """
+
+    NAT_TYPE: ClassVar[ht.Type] = NumericType(NumericType.Kind.Nat).to_hugr()
+
+    def __init__(self, op_name: str) -> None:
+        self.op_name = op_name
+
+    def nat_to_usize(self, value: Wire) -> Wire:
+        op = ops.Custom(
+            name="itousize",
+            signature=ht.FunctionType([self.NAT_TYPE], [ht.USize()]),
+            extension="arithmetic.conversions",
+            args=[ht.BoundedNatArg(NumericType.INT_WIDTH)],
+        )
+        return self.builder.add_op(op, value)
+
+    def usize_to_nat(self, value: Wire) -> Wire:
+        op = ops.Custom(
+            name="ifromusize",
+            signature=ht.FunctionType([self.NAT_TYPE], [ht.USize()]),
+            extension="arithmetic.conversions",
+            args=[ht.BoundedNatArg(NumericType.INT_WIDTH)],
+        )
+        return self.builder.add_op(op, value)
+
+    def compile(self, args: list[Wire]) -> list[Wire]:
+        sig = ht.FunctionType(self.ty.input.copy(), self.ty.output.copy())
+        for i, ty in enumerate(sig.input):
+            if ty == self.NAT_TYPE:
+                args[i] = self.nat_to_usize(args[i])
+                sig.input[i] = ht.USize()
+        op = ops.Custom(self.op_name, sig, extension="guppy.unsupported", args=[])
```

Comment:
No, the validator just treats them as opaque ops if it doesn't know the extension. 

We need to wait until #454 is merged to get the latest angle extension, otherwise the validator will complain about missing ops like `afromrad` from https://github.com/CQCL/tket2/pull/581.